src/page-multi-facility/ReSyncRefFacilityModal.tsx

```diff
@@ -0,0 +1,266 @@
+  syncedRefFacilities: FacilityReferenceMapping,
+) => {
+  return (facility: Facility) => {
+    return !Object.keys(syncedRefFacilities).includes(facility.id);
+  };
+};
+
+interface ModalProps {
+  open: boolean;
+  stateName: ModelInputs["stateName"];
+  systemType: Facility["systemType"];
+  onClose: () => void;
+}
+
+const ReSyncRefFacilityModal: React.FC<ModalProps> = ({
+  open,
+  stateName,
+  systemType,
+  onClose,
+}) => {
+  const rejectionToast = useRejectionToast();
+  const [scenarioState, dispatchScenarioUpdate] = useScenario();
+  const {
+    state: { facilities: facilitiesState, referenceFacilities },
+  } = useFacilities();
+
+  const [selections, setSelections] = useState<{
+    [refFacilityId: string]: Facility["id"];
+  }>({});
+
+  const scenario = scenarioState.data;
+  const syncedRefFacilities = scenario?.[referenceFacilitiesProp] || {};
+
+  if (!open) return null;
+
+  const facilities = Object.values(facilitiesState).filter(
+    filterSyncedFacilities(syncedRefFacilities),
+  );
+
+  async function handleSave() {
+    if (Object.keys(selections)) {
+      const facilityIdToReferenceId = invert(selections);
+
+      await rejectionToast(
+        saveScenario({
+          ...scenario,
+          [referenceFacilitiesProp]: Object.assign(
+            {},
+            scenario?.[referenceFacilitiesProp],
+            facilityIdToReferenceId,
+          ),
+        }).then((savedScenario) => {
+          if (savedScenario) dispatchScenarioUpdate(savedScenario);
+        }),
+      );
+    }
+    onClose();
+  }
+
+  return (
+    <ModalDialog
+      open={open}
+      title={<SyncDataTitle stateName={stateName} systemType={systemType} />}
+    >
+      <ModalContent>
+        <HeaderRow>
+          <Header>Facilities with available prepopulated data</Header>
+          <Header>Your facilities</Header>
+        </HeaderRow>
+        {Object.values(referenceFacilities).map((refFacility) => {
+          return (
+            <Row key={refFacility.id}>
+              <FacilityName>{refFacility.canonicalName}</FacilityName>
+              <FacilitiesSelect
+                selections={selections}
+                facilities={facilities}
+                value={selections[refFacility.id]}
+                onChange={(facilityId) => {
+                  if (facilityId) {
+                    setSelections({
+                      ...selections,
+                      [refFacility.id]: facilityId,
+                    });
+                  } else {
+                    const selectionsCopy = { ...selections };
+                    delete selectionsCopy[refFacility.id];
+                    setSelections(selectionsCopy);
+                  }
+                }}
+              />
+            </Row>
+          );
+        })}
+      </ModalContent>
+      <ModalFooter>
+        <CancelButton onClick={onClose}>Not now</CancelButton>
+        <SaveButton
+          disabled={!Object.keys(selections).length}
```

Comment:
I can't click on the Save button if I don't make any selections, but it might be nice to visually gray the button out as well.